pydatastructs/linear_data_structures/algorithms.py

```diff
@@ -233,3 +234,69 @@ def brick_sort_parallel(array, num_threads, **kwargs):
 
     if _check_type(array, DynamicArray):
         array._modify(force=True)
+
+def _matrix_multiply_helper(m1, m2, row, col):
+    s = 0
+    for i in range(len(m1)):
+        s += m1[row][i] * m2[i][col]
+    return s
+
+def matrix_multiply_parallel(matrix_1, matrix_2, num_threads):
+    """
+    Implementes concurrent Matrix multiplication
+
+    Parameters
+    ==========
+
+    matrix_1: list of list / tuple of tuple / OneDimensionalArray of OneDimensionalArray
+        Left matrix
+
+    matrix_2: Any matrix representation
+        Right matrix
+
+    num_threads: int
+        The maximum number of threads
+        to be used for multiplication.
+
+    Raises
+    ======
+
+    IndexError
+        When the columns in matrix_1 are not equal to the rows in matrix_2
+
+    Returns
+    =======
+
+    C: list
+        The result of matrix multiplication.
+
+    Examples
+    ========
+
+    >>> from pydatastructs import matrix_multiply_parallel
+    >>> I = [[1, 1, 0], [0, 1, 0], [0, 0, 1]]
+    >>> J = [[2, 1, 2], [1, 2, 1], [2, 2, 2]]
+    >>> matrix_multiply_parallel(I, J, num_threads = 5)
+    [[3, 3, 3], [1, 2, 1], [2, 2, 2]]
+
+    References
+    ==========
+    .. [1] https://en.wikipedia.org/wiki/Matrix_multiplication_algorithm#Parallel_and_distributed_algorithms
```

Comment:
This link refers to an algorithm which is not relevant to the implementation here. 
```suggestion
    .. [1] https://www3.nd.edu/~zxu2/acms60212-40212/Lec-07-3.pdf
```